RetrospectiveExtension.Frontend/dal/azureDevOpsWorkItemService.tsx

```diff
@@ -85,7 +90,8 @@ class WorkItemService {
       },
     ];
 
-    return this._httpClient.createWorkItem(operation, VSS.getWebContext().project.id, WorkItemService.retrospective_type);
+    return getProjectId()
+      .then(projectId => this._httpClient.createWorkItem(operation, projectId, WorkItemService.retrospective_type));
```

Comment:
curious why you are using `then` here instead of making the method async and awaiting. any limitations here?

Reply:
This file actually contains both `async` and `.then` styles. @polatengin what's your thoughts? If we're going to change to async, it'll make the code more readable so something to consider for future PR?

Reply:
I'd like to have a single approach, most of the codebase is using `async-await` pattern.

if we can make the method `async`, we should definitely convert this to `await`, instead of using `.then()` functions.

sometimes, we can't convert method to `async` method, in those cases, we can use `.then()` functions.

Reply:
Converted several methods that use `getProjectId` to _async_ type. For the rest of the file, we should fix them in a separate PR since it's out of scope for this one.